### PR TITLE
hmem: Support dev reg with FI_HMEM_ZE

### PR DESCRIPTION
**Pull request by iziemba**
Load/store ZE buffer is done by mmaping the DMA buf FD.

### CODE REVIEW

**Comment by j-xiong** [comment]
Intel CI error:

```
10:21:39  src/hmem_ze.c: In function ‘ze_dma_buf_sync’:
10:21:39  src/hmem_ze.c:1233:9: warning: implicit declaration of function ‘ioctl’ [-Wimplicit-function-declaration]
10:21:39     ret = ioctl(fd, DMA_BUF_IOCTL_SYNC, sync);
10:21:39           ^~~~~
10:21:39  In file included from src/hmem_ze.c:46:0:
10:21:39  src/hmem_ze.c:1233:19: error: expected expression before ‘struct’
10:21:39     ret = ioctl(fd, DMA_BUF_IOCTL_SYNC, sync);
10:21:39                     ^
```

File: src/hmem_ze.c [hunk 1]
```diff
@@ -1119,6 +1120,188 @@ int ze_hmem_host_unregister(void *ptr)
...
+		ret = -FI_EIO;
+		goto err_free_handle;
+	}
+
+	dev_handle->base_dev = ze_base_dev;
+	dev_handle->base_host = ze_mmap_addr;
+	dev_handle->len = ze_base_size;
+	dev_handle->fd = fd;
+
+	*handle = (uint64_t) dev_handle;
```
Comment (shefty) [inline src/hmem_ze.c:1199]: Need to use (uintptr_t) in conversions between points and integer types

File: src/hmem_ze.c [hunk 2]
```diff
@@ -1119,6 +1120,188 @@ int ze_hmem_host_unregister(void *ptr)
...
+err:
+	return ret;
+}
+
+int ze_dev_unregister(uint64_t handle)
+{
+	struct ze_dev_reg_handle *dev_handle;
+	int ret;
+
+	dev_handle = (struct ze_dev_reg_handle *) handle;
```
Comment (shefty) [inline src/hmem_ze.c:1214]: (uintptr_t)

**Review by j-xiong — APPROVED** [review]

**Comment by j-xiong** [comment]
@shijin-aws Any insight into the AWS CI failure? I haven't seen a success recently.

**Comment by shijin-aws** [comment]
bot:aws:retest

**Comment by iziemba** [comment]
bot:aws:retest

**Pull request merged**